compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -371,14 +466,29 @@ object InstrumentCoverage:
   val name: String = "instrumentCoverage"
   val description: String = "instrument code for coverage checking"
 
-  /** Extractor object for trees produced by `insertInvokeCall`. */
-  object InstrumentedBlock:
-    private def isInvokedCall(app: Apply)(using Context): Boolean =
-      app.span.isSynthetic && app.symbol == defn.InvokedMethodRef.symbol
-
-    def unapply(t: Tree)(using Context): Option[(Apply, Tree)] =
-      t match
-        case Block((app: Apply) :: Nil, expr) if isInvokedCall(app) =>
-          Some((app, expr))
-        case _ =>
-          None
+  /**
+   * An instrumented Tree, in 3 parts.
+   * @param pre preparation code, e.g. lifted arguments. May be empty.
+   * @param invokeCall call to Invoker.invoked(dir, id), or an empty tree.
+   * @param expr the instrumented expression, executed just after the invokeCall
+   */
+  case class InstrumentedParts(pre: List[Tree], invokeCall: Apply | Thicket, expr: Tree):
+    require(pre.isEmpty || (pre.nonEmpty && !invokeCall.isEmpty), "if pre isn't empty then invokeCall shouldn't be empty")
+
+    /** Turns this into an actual Tree. */
+    def toTree(using Context): Tree =
+      if invokeCall.isEmpty then expr
+      else if pre.isEmpty then Block(invokeCall :: Nil, expr)
+      else Block(pre :+ invokeCall, expr)
+
+    /** Applies a function on the instrumented expression. */
+    def map(f: Tree => Tree): InstrumentedParts =
```

Comment:
Is this method used anywhere?

Reply:
It's not! thanks for pointing this out, I'll remove it